addons/stock_dropshipping/models/stock.py

```diff
@@ -21,6 +21,11 @@ def _get_partner_id(self, values, rule):
             return False
         return super()._get_partner_id(values, rule)
 
+    def _get_picking_type_code_domain(self):
+        if self.action == 'buy':
+            return [('code', 'in', ['incoming', 'dropship'])]
```

Comment:
It won't work because stock_dropshipping and purchase_stock have parallel dependencies. I would do 
get domain by action like
```
in purchase_stock
domain = type_domain_by_action.get('buy', Domain())
domain = Domain.AND(domain, Domain([('code', '=', 'incoming')])
type_domain_by_action['buy'] = domain
```